README.txt

```diff
@@ -49,14 +49,11 @@ the developed codebase::
 Known issues
 ------------
 
-* No Python-3 support yet.
-
-* Case like::
+* Python-2 support will be dropped in the next major version.
```

Comment:
Oh, you do intend to drop Python 2 support!  I did not expect that.

Reply:
The reason being: the Python 3 version will use the `ast` module. Even though Python 2 has the `ast` module too, the tree structure is different between Python 2 and Python 3. In informal testing I can see those differences do not affect what `importchecker` is interested in, but I'm not sure I can guarantee that nor  am I sure how to make the codebase such that it can deal with both tree structures. There's also a time constraint on my side :/

Reply:
I'm the maintainer of https://github.com/mgedmin/findimports which uses the `ast` module to, essentially, do what you're doing here in importchecker.

I support both Python 2 and Python 3 from a single codebase.  I do not remember that presenting any difficulties.

The difficult part was switching from the `compiler` module to the `ast` module (on both Python 2 and Python 3): https://github.com/mgedmin/findimports/commit/96f0f2386b01b5265ab36142aa05d4f3f4bd956f

Reply:
@mgedmin thanks for the review and comments. `findimports` seems interesting. I wonder if `importchecker`could be replaced by it. Still, your "switch" to `ast` looks very familiar to what I did in my Python 3 branch. I was a bit put off claiming Python 2 *and* 3 support in the same code base when reading up on the differences in the syntax trees between the two (for example here http://joao.npimentel.net/2015/07/23/python-2-vs-python-3-ast-differences).

That said, I could tone down the "I'm dropping Python 2"-statement and say "This seems to still work in Python 2." (as informal testing thus far indeed seems to indicate).

Reply:
`findimports` has known bugs I haven't had time to fix for years (e.g. it doesn't understand code like `import foo.bar; foo.bar.baz()`).  Today I use `flake8` to discover unused imports.

The "Seems to still work in Python 2" bit could be easily verified with [tox](https://pypi.python.org/pypi/tox) or Travis CI, if you trust your test suite ;)

Reply:
I don't trust my test suite TBH. I added it after the fact for code someone else wrote years and years ago :) Still, I added the test suite specifically to have some sort of baseline for the Python 3 port. I'll add a tox and travis config. We use `flake8` too, however `importchecker` catches certain import issues that `flake8` does not.